### PR TITLE
various python3 related fixes

### PR DESCRIPTION
**Pull request by eviljeff**
part of #10413 - more needed.

### CODE REVIEW

File: src/olympia/amo/views.py [hunk 1]
```diff
@@ -93,7 +93,7 @@ def handler404(request):
 
 @non_atomic_requests
 def handler500(request):
-    if request.is_legacy_api:
+    if getattr(request, 'is_legacy_api', False):
```
Comment (diox) [inline src/olympia/amo/views.py]: Why is this necessary?

Comment (eviljeff) [reply src/olympia/amo/views.py]: It probably isn't in the end - there's an exception thrown, seemingly before the `is_legacy_api` property is added to `request`, and without this the real exception is replaced with one about `is_legacy_api` being undefined..

Comment (diox) [reply src/olympia/amo/views.py]: ok, it kinda makes sense anyway...
